packages/framework-provider-local/src/library/read-model-adapter.ts

```diff
@@ -42,9 +43,20 @@ export async function storeReadModel(
   _config: BoosterConfig,
   logger: Logger,
   readModelName: string,
-  readModel: ReadModelInterface
+  readModel: ReadModelInterface,
+  expectedCurrentVersion: number
 ): Promise<void> {
-  await db.store({ typeName: readModelName, value: readModel })
+  logger.debug('[ReadModelAdapter#storeReadModel] Storing readModel ' + JSON.stringify(readModel))
+  try {
+    await db.store({ typeName: readModelName, value: readModel } as ReadModelEnvelope)
+  } catch (e) {
+    // The error will be thrown, but in case of a conditional check, we throw the expected error type by the core
+    // TODO: verify the name of the exception thrown in Local Provider
+    if (e.name == 'TODO') {
+      throw new OptimisticConcurrencyUnexpectedVersionError(e.message)
+    }
```

Comment:
TODO. Catch the exception raised when store operation failed, just to implement Optimistic Concurrency. This is not part of the MVP